src/Raven.Client/ServerWide/Operations/Certificates/CertificateDefinition.cs

```diff
@@ -4,18 +4,35 @@
 
 namespace Raven.Client.ServerWide.Operations.Certificates
 {
-    public class CertificateDefinition
+    public class CertificateDefinition : CertificateMetadata
     {
-        public string Name;
         public string Certificate;
         public string Password;
+        public string PublicKeyPinningHash;
+        public string CollectionPrimaryKey = string.Empty;
+        public List<string> CollectionSecondaryKeys = new List<string>();
```

Comment:
What is that?

Reply:
Do you ask about `CollectionSecondaryKeys`?
As I understand it if the user supplies PFX that contains more than one certificate we save each one separately but we present all of them as secondary except to the first.

But your question makes me think about how the decision here should be.
The idea here is to give the user sufficient information to call for the edit but the original problem was to prevent sending sensitive data through the wire so maybe it would make more sense to divide it by sensitivity?
So if we want to keep `CertificateDefinition` unchanged to not break anything:
```
public class CertificateDefinition : InsensitiveCertificateDefinition
{
    public string Certificate;
    public string Password;

    public DynamicJsonValue ToJson(bool metadataOnly = false)
    {
        var jsonValue = base.ToJson();
        if (metadataOnly == false)
        {
            jsonValue[nameof(Certificate)] = Certificate;
        }
        return jsonValue;
    }
}

public class InsensitiveCertificateDefinition
{
    public string Name;
    public SecurityClearance SecurityClearance;
    public string Thumbprint;
    public DateTime? NotAfter;
    public Dictionary<string, DatabaseAccess> Permissions = new Dictionary<string, DatabaseAccess>(StringComparer.OrdinalIgnoreCase);
    public List<string> CollectionSecondaryKeys = new List<string>();
    public string CollectionPrimaryKey = string.Empty;
    public string PublicKeyPinningHash;

    public DynamicJsonValue ToJson()
    {
        var permissions = new DynamicJsonValue();
        
        if (Permissions != null)
            foreach (var kvp in Permissions)
                permissions[kvp.Key] = kvp.Value.ToString();

        var jsonValue = new DynamicJsonValue
        {
            [nameof(Name)] = Name,
            [nameof(Thumbprint)] = Thumbprint,
            [nameof(SecurityClearance)] = SecurityClearance,
            [nameof(Permissions)] = permissions,
            [nameof(NotAfter)] = NotAfter,
            [nameof(CollectionSecondaryKeys)] = CollectionSecondaryKeys,
            [nameof(CollectionPrimaryKey)] = CollectionPrimaryKey,
            [nameof(PublicKeyPinningHash)] = PublicKeyPinningHash;
        };
        return jsonValue;
    }
}
```

Reply:
I think that there is some issues with the reason for this feature.
This is _not_ about security. We assume the following:

* The connection is encrypted between us and the server.
* The certificate we keep is the public info only.

This is mostly so we'll not need to download and send back the certificate on each time. Users want to be able to provide a way to edit permissions on a user using just the thumbprint of the certificate.

Reply:
But the user still needs to get the old configuration so he could edit it and resend to the `edit` endpoint.
So it is all about the size of the request/response here?

Reply:
I think that this is mostly about the ease of use.
What if we change things up.
Add an `EditPermissions` operation that would allow simply replacing the permissions by thumbprint?

Reply:
Without updating anything else?

Reply:
I think that this operation is already lightweight enough. Why making it more complex, adding an endpoint or any other additional logic?

```
string thumbprint, string name, Dictionary<string, DatabaseAccess> permissions, SecurityClearance clearance
```

Reply:
So if it is not about sensitive data maybe I will remove the `metadataOnly` if it is not necessary (include the division to CertificateDefinition & CertificateMetadata)?

Reply:
No